client/lib/signup/flow-controller.js

```diff
@@ -130,6 +130,45 @@ assign( SignupFlowController.prototype, {
 		}.bind( this ) );
 	},
 
+	/**
+	 * This assertion helper validates that there are dependencies for the step provided
+	 * by completed steps
+	 * @param step The step to validate
+	 */
+	_assertStepDependencies( step ) {
+		const dependencies = steps[ step.stepName ].dependencies || [];
+		const dependenciesFound = pick( SignupDependencyStore.get(), dependencies );
+		const dependenciesSatisfied = dependencies.length === keys( dependenciesFound ).length;
+		const signupProgress = filter(
+			SignupProgressStore.get(),
+			_step => ( -1 !== this._flow.steps.indexOf( _step.stepName ) )
+		);
+
```

Comment:
I think a lot of this logic is shared with `_canProcessStep`. It might be worth removing it there.